src/network/networkd-ndisc.c

```diff
@@ -199,7 +199,12 @@ static int ndisc_request_route(Route *in, Link *link, sd_ndisc_router *rt) {
                 r = sd_ndisc_router_get_hop_limit(rt, &hop_limit);
                 if (r < 0 && r != -ENODATA)
                         return log_link_warning_errno(link, r, "Failed to get default router hop limit from RA: %m");
-        }
+
+                link->network->ipv6_hop_limit = hop_limit;
+                r = link_set_ipv6_hop_limit(link);
```

Comment:
According to https://www.rfc-editor.org/rfc/rfc4861.html#section-6.3.4
>    If the received Cur Hop Limit value is non-zero, the host SHOULD set
   its CurHopLimit variable to the received value.

so networkd shouldn't touch it if Cur Hop Limit is 0